repl: deprecate unused function convertToContext

princejwesley opened the pull request:
<!--
Thank you for your pull request. Please review below requirements.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test nosign` (Windows) passes
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
repl

##### Description of change
<!-- Provide a description of the change below this comment. -->
Remove unused function `convertToContext`

thefourtheye commented:
cc @ChALkeR Can you please search for this function in the npm modules?

addaleax commented:
Good idea, LGTM if nothing too relevant comes up (I’d be surprised if it did, this code is unused since 2010).

CI: https://ci.nodejs.org/job/node-test-commit/4216/

Trott commented:
LGTM. Probably worth a CITGM run too, just to flush out anything obvious early on:

https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/341/

Hopefully I did that correctly....

cjihrig commented:
LGTM if it's not in use.

Fishrock123 commented:
Is this supposed to be a user API?

princejwesley commented:
@Fishrock123 No.

###### History
- Introduced [here](https://github.com/nodejs/node/commit/6915188a461b455e7112634f7a72612a9c6de54d#diff-b13d72249263845d8e8341db0426f9d3R122) as `convertToScope` and used inside [`readline`](https://github.com/nodejs/node/commit/6915188a461b455e7112634f7a72612a9c6de54d#diff-b13d72249263845d8e8341db0426f9d3R39) function  (**Sep 25, 2009**)

- New context is used instead of [`convertToScope`](https://github.com/nodejs/node/commit/82813f2f8cfb3fae36e8e5e0ea9aeb83650aa253#diff-b13d72249263845d8e8341db0426f9d3L64) (**May 10, 2010**)

- Renamed [convertToScope -> convertToContext](https://github.com/nodejs/node/commit/242161bef2e3631d315f8da8a62c088bdc100c87#diff-b13d72249263845d8e8341db0426f9d3R190) (**Jun 24, 2010**)

ChALkeR commented:
@thefourtheye Does not seem to be used anywhere (based on the old data from 2016-01-28, new data will be ready a bit later).

jasnell commented:
Even tho this doesn't appear to be used, I would still recommend (and would be more comfortable with) a proper deprecation cycle before removing outright.

thefourtheye commented:
Considering my recent attempts with `fs`, I am inclined to go with @jasnell's comment.

princejwesley commented:
> ... a proper deprecation cycle before removing outright

@jasnell  Using [util#deprecation](https://nodejs.org/api/util.html#util_util_deprecate_function_string) ?

jasnell commented:
We have two methods: soft deprecation (docs only) or util.deprecation. We
should likely start with soft deprecation in v7.

On Tuesday, July 26, 2016, Prince John Wesley <notifications@github.com>
wrote:

> ... a proper deprecation cycle before removing outright
>
> @jasnell <https://github.com/jasnell> Using util#deprecation
> <https://nodejs.org/api/util.html#util_util_deprecate_function_string> ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node/pull/7829#issuecomment-235319300>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAa2eTYlUF2uNPYwwtRiIyKQjNzB5ED2ks5qZjG8gaJpZM4JSSl3>
> .
>

princejwesley commented:
@jasnell FYI, `convertToContext` method is not currently documented in `repl.md`.

addaleax commented:
Yeah, soft deprecation isn’t really applicable here, and `util.deprecate` seems like the way to go.

princejwesley commented:
@addaleax @jasnell We need a clean up/todo label for v8.x. Do we have one?

thefourtheye commented:
@princejwesley Just a leave a to-do like this `TODO(princejwesley):` followed by the note.

yorkie commented on lib/repl.js:
```diff
@@ -1222,7 +1222,7 @@ REPLServer.prototype.convertToContext = function(cmd) {
 
   return cmd;
-};
+}, 'replServer.convertToContext will be removed in v8.0.0');
```
use "after", that represents a range, instead of the precise word "in" would be better?

princejwesley replied:
@yorkie `removed after v7` ?

yorkie replied:
Exactly what I mean :-) @princejwesley

yorkie commented:
LGTM

addaleax commented:
LGTM, CI: https://ci.nodejs.org/job/node-test-commit/4305/

jasnell commented on lib/repl.js:
```diff
@@ -1222,7 +1224,7 @@ REPLServer.prototype.convertToContext = function(cmd) {
 
   return cmd;
-};
+}, 'replServer.convertToContext will be removed after v7');
```
Our deprecation messages typically take a form like, 'replServer.convertToContext() is deprecated', without committing to a specific version when it may be removed. The message should also indicate if there is an alternative that the developer should use.

yorkie commented:
Oops, it might need a test case.

jasnell commented:
@nodejs/ctc
LGTM

princejwesley commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/3468/

yorkie commented:
LGTM if CI is green :-)

jasnell commented:
@nodejs/ctc ... putting out a last call for comments on this. If there are no objections to adding this deprecation I will land tomorrow.

jasnell commented:
Ok, it took longer than a day but landed in 488d28d! :-)

refack commented:
Just found this. Can we remove it?
